cluster/message.proto

```diff
@@ -22,14 +27,26 @@ message Command {
         command.ExecuteRequest execute_request = 2;
         command.QueryRequest query_request = 3;
     }
+
+    Credentials credentials = 4;
+}
+
+message ResultsError {
+    enum Type {
+        RESULTS_ERROR_UNSPECIFIED = 0;
+        RESULTS_ERROR_UNAUTHORIZED = 1;
+        RESULTS_ERROR_NIL_QUERY_REQUEST =2;
+    }
+    Type type = 1;
+    string error = 2;
 }
 
 message CommandExecuteResponse {
-	string error = 1;
+	ResultsError error = 1;
```

Comment:
You can't do this, it's not backwards-compatible -- it would prevent this software running with, say, nodes running 7.5. You need to add a new field. I suggest a simple Boolean field to `CommandExecuteResponse` and `CommandQueryResponse` , indicating the "unauthorized" condition. 

It means you'll need to check two conditions -- `Error != ""` or "Unauthorized`. If either is true, return an error.